internal/backend/sema/backend.go

```diff
@@ -0,0 +1,99 @@
+package sema
+
+import (
+	"context"
+	"io"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/restic/restic/internal/debug"
+	"github.com/restic/restic/internal/errors"
+	"github.com/restic/restic/internal/restic"
+)
+
+// make sure that SemaphoreBackend implements restic.Backend
+var _ restic.Backend = &SemaphoreBackend{}
+
+// SemaphoreBackend limits the number of concurrent operations.
+type SemaphoreBackend struct {
+	restic.Backend
+	sem semaphore
+}
+
+// New returns a new backend that saves all data in a map in memory.
+func New(be restic.Backend) *SemaphoreBackend {
```

Comment:
sema.New doesn't quite express what this function does. NewBackend? Wrap? Limit?

Reply:
The dry run backend and the RetryBackend also use a `New` function. I would settle for `NewBackend` although that adds a certain inconsistency regarding the existing backends.